src/structures/APIMessage.js

```diff
@@ -151,8 +151,8 @@ class APIMessage {
 
     let nonce;
     if (typeof this.options.nonce !== 'undefined') {
-      nonce = parseInt(this.options.nonce);
-      if (isNaN(nonce) || nonce < 0) throw new RangeError('MESSAGE_NONCE_TYPE');
+      nonce = this.options.nonce;
+      if ((typeof nonce === 'number' && !Number.isInteger(nonce)) || !nonce) throw new RangeError('MESSAGE_NONCE_TYPE');
```

Comment:
This logic does not seem sound to me, some valid values are currently failing here, yet some invalid values pass.

For a false positive example: `0` and `''` fail here, but would pass on Discord's end (although they seem to just drop them)
For a false negative example: `{}` and `[]` pass here but fails on Discord's end.